code/game/objects/items/weapons/soap.dm

```diff
@@ -23,12 +24,27 @@
 	if(target == user && user.a_intent == INTENT_GRAB && ishuman(target))
 		var/mob/living/carbon/human/muncher = user
 		if(muncher && isdrask(muncher))
-			to_chat(user, "<span class='notice'>You take a bite of [src]. Delicious!</span>")
+			times_eaten += 1
 			playsound(user.loc, 'sound/items/eatfood.ogg', 50, 0)
 			user.adjust_nutrition(2)
-		return
+			if(times_eaten <= 3)
```

Comment:
Try to avoid magic numbers like this. Either make a define or add it as a variable on the item itself.